sympy/functions/special/delta_functions.py

```diff
@@ -208,22 +227,42 @@ def fdiff(self, argindex=1):
             raise ArgumentIndexError(self, argindex)
 
     @classmethod
-    def eval(cls, arg):
+    def eval(cls, arg, H0=None):
         arg = sympify(arg)
-        if arg is S.NaN:
-            return S.NaN
-        elif fuzzy_not(im(arg).is_zero):
-            raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
-        elif arg.is_negative:
+        if arg.is_negative:
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero:
+            if H0 == S.NaN:
+                return cls(arg)
+            elif H0 != None:
```

Comment:
Here I feel that replacing `elif H0 != None:` with `elif H0 == 1/2 or H0 == 1:` would be more appropriate.
and raise an error for other values of H0.